### PR TITLE
Fix script waits matching stale terminal output

### LINKED ISSUE #1821: [Bug] 自动化脚本会匹配已经输出到终端的内容
### Operating system

Windows

### Netcatty version

1.1.50

### How did you install Netcatty?

GitHub Release (.dmg / .exe / .AppImage / .deb / .rpm / .pacman)

### Affected area

Snippets

### Can you reproduce it?

Always (100%)

### Steps to reproduce

1. 配置一个根据终端输出内容执行特定动作的脚本或者输出匹配时运行的脚本
2. 运行脚本，同时在终端制造对开关键字打印
3. 脚本捕获到对应输出后执行相应动作正常
4. 重新执行脚本，但是不在制造关键字输出
5. 脚本依然能匹配上面已经输出过的关键字
6. 通过回车或者清屏后终端没有关键字脚本又恢复正常

https://github.com/user-attachments/assets/f873c30f-129f-4bbe-b636-0449a312f35e

### Expected behavior

应该只对输出内容做匹配，输出后的内容不在匹配的范围内

### Actual behavior

只要终端窗口内的内容匹配到关键字后脚本都会执行动作

### Logs / screenshots

Reported from Netcatty Settings (netcatty 1.1.50).

User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) netcatty/1.1.50 Chrome/148.0.7778.218 Electron/42.3.3 Safari/537.36

### Before submitting

- [x] I searched existing issues and did not find a duplicate
- [x] I removed passwords, private keys, and other secrets from this report

### PR DESCRIPTION
**Pull request by binaricat**
Fixes #1821

## Root cause
Script startup synchronization copied the currently visible terminal screen into the session output buffer, but normal `waitFor` / `waitForAny` treated that copied snapshot as fresh tail output. If a keyword was still visible from a previous run, the next run could match it immediately without any new terminal output.

## Fix
- Mark the current output buffer as consumed after startup screen synchronization.
- Preserve only the current shell prompt for the explicit `waitForPrompt` startup use case.
- Keep normal `waitFor` / `waitForAny` scoped to output produced after the script starts or after new terminal data arrives.
- Add regression coverage for stale keyword matching, prompt preservation, and normal `waitForAny` not consuming preserved startup prompts.

## Verification
- `node --test --import tsx electron/scripts/sessionOutputBuffer.test.cjs`
- `node --test --import tsx electron/bridges/scriptBridge.test.cjs`
- `node --test --import tsx electron/scripts/scriptRuntime.test.cjs`
- `npm run lint`
- `npm test` (3794 tests: 3791 passed, 3 skipped, 0 failed)

## Local review
Two local review agents were run. The first found an important edge case where generic `waitForAny` could still consume a preserved startup prompt; that was fixed. The second review was clean with no blocker or important findings.

### CODE REVIEW

File: electron/bridges/scriptBridge.cjs [hunk 1]
```diff
@@ -281,7 +283,11 @@ function syncOutputBufferFromSnapshot(sessionId) {
     if (!existing.includes(tail.trim())) {
       buffer.append(tail.startsWith("\n") ? tail : `\n${tail}`);
     }
-  }).catch(() => {});
+  } catch {
+    // Keep startup synchronization best-effort; the current buffer is still baselined below.
+  } finally {
+    buffer.markCurrentOutputConsumed({ preserveTailPatterns: shellPromptPatterns() });
```
[the remark this answers is not in the document]
Comment (binaricat) [reply electron/bridges/scriptBridge.cjs]: Fixed in c10debea. Startup sync now records the output buffer contents before awaiting the screen snapshot and only marks that pre-sync range (plus any snapshot text appended when no live output arrived) as consumed. If terminal data arrives while the snapshot request is pending, it remains pending for the first script wait. Added regression coverage in `script waitFor sees output that arrives while startup snapshot sync is pending`.

**Comment by binaricat** [comment]
@codex review
